docs(iam): resource and data_source (#1502)
scaleway/terraform-provider-scaleway: docs/resources/iam_api_key.md

---
page_title: "Scaleway: scaleway_iam_api_key"
description: |-
Manages Scaleway IAM API Keys.
---

# scaleway_iam_api_key

| WARNING: This resource is in beta version. If your are in the beta group, please set the variable `SCW_ENABLE_BETA=true` in your `env` in order to use this resource. |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|

Creates and manages Scaleway IAM API Keys. For more information, please check [the documentation](https://developers.scaleway.com/en/products/iam/api/v1alpha1/#api-keys-3665ae)

## Example Usage

```hcl

> resource "scaleway_iam_application" "main" {

Reviewer: Maybe try different naming

```suggestion
resource "scaleway_iam_application" "ci_cd" {
```

|------------------------------------------------------------------------------------------------------------------------------------------------------------------|

Creates and manages Scaleway IAM API Keys. For more information, please check [the documentation](https://developers.scaleway.com/en/products/iam/api/v1alpha1/#api-keys-3665ae)

## Example Usage

```hcl
resource "scaleway_iam_application" "main" {
    name = "My application"
}

resource "scaleway_iam_api_key" "main" {
    application_id = scaleway_iam_application.main.id
    description = "a description"
}
```

## Arguments Reference

The following arguments are supported:

- `description`: (Optional) The description of the iam api key.
- `application_id`: (Optional) ID of the application attached to the api key.
  Only one of the `application_id` and `user_id` should be specified.
- `user_id` - (Optional) ID of the user attached to the api key.
  Only one of the `application_id` and `user_id` should be specified.
- `expires_at` - (Optional) The date and time of the expiration of the iam api key. Please note that in case of change, the resource will be recreated.

> - `default_project_id` - (Optional)The project ID that is used as a default value for all resources

Reviewer: Need better wording. I suggest this for now but we might add link to a better doc page for the GA (cc @crlptl)

```suggestion
- `default_project_id` - (Optional) The default project ID to use with object storage.
```

## Attributes Reference

In addition to all above arguments, the following attributes are exported:

- `created_at` - The date and time of the creation of the iam api key.

> - `updated_at` - The date and time of the last update of the iam api key.

Reviewer: In general I don't think it is relevant to have these 2 fields in reference.

Reviewer 2: They should be referenced when present in schema, not all resources have those fields
